planner/core/task.go

```diff
@@ -1709,7 +1709,13 @@ func BuildFinalModeAggregation(
 				// we must call deep clone in this case, to avoid sharing the arguments.
 				sumAgg := aggFunc.Clone()
 				sumAgg.Name = ast.AggFuncSum
-				sumAgg.RetTp = partial.Schema.Columns[partialCursor-1].GetType()
+				err := sumAgg.TypeInfer(sctx)
```

Comment:
Does not need to infer type for `count` function here?

Reply:
we can call inferType to align types in any case.